forwarder/update_handlers/ep01_serialiser.py

```diff
@@ -0,0 +1,97 @@
+
+from forwarder.update_handlers.schema_serialisers import CASerialiser, PVASerialiser
+
+
+def _serialise(
+    source_name: str, conn_status: ConnectionInfo, timestamp_ns: int
+) -> Tuple[bytes, int]:
+    return (
+        serialise_ep01(
+            timestamp_ns=timestamp_ns,
+            status=conn_status,
+            source_name=source_name,
+        ),
+        timestamp_ns,
+    )
+
+
+class ep01_CASerialiser(CASerialiser):
+    def __init__(self, source_name: str):
+        self._source_name = source_name
+        self._conn_status: ConnectionInfo = ConnectionInfo.NEVER_CONNECTED
+
+    def serialise(
+        self, update: CA_Message, **unused
+    ) -> Union[Tuple[bytes, int], Tuple[None, None]]:
+        return None, None
+
+    def conn_serialise(
+        self, pv: str, state: str
+    ) -> Tuple[Optional[bytes], Optional[int]]:
+        from forwarder.kafka.kafka_helpers import seconds_to_nanoseconds
+
+        state_str_to_enum: Dict[str, ConnectionInfo] = {
+            "connected": ConnectionInfo.CONNECTED,
+            "disconnected": ConnectionInfo.DISCONNECTED,
+            "destroyed": ConnectionInfo.DESTROYED,
+            "cancelled": ConnectionInfo.CANCELLED,
+            "finished": ConnectionInfo.FINISHED,
+            "remote_error": ConnectionInfo.REMOTE_ERROR,
+        }
+        self._conn_status = state_str_to_enum.get(state, ConnectionInfo.UNKNOWN)
+        return _serialise(
+            self._source_name, self._conn_status, seconds_to_nanoseconds(time.time())
+        )
+
+    def start_state_serialise(self):
+        from forwarder.kafka.kafka_helpers import seconds_to_nanoseconds
+
+        return _serialise(
+            self._source_name, self._conn_status, seconds_to_nanoseconds(time.time())
+        )
+
+
+class ep01_PVASerialiser(PVASerialiser):
+    def __init__(self, source_name: str):
+        self._source_name = source_name
+        self._conn_status: ConnectionInfo = ConnectionInfo.NEVER_CONNECTED
+
+    def serialise(
+        self, update: Union[p4p.Value, RuntimeError], **unused
+    ) -> Union[Tuple[bytes, int], Tuple[None, None]]:
+        from forwarder.kafka.kafka_helpers import seconds_to_nanoseconds
+
+        if isinstance(update, p4p.Value):
+            timestamp = (
+                update.timeStamp.secondsPastEpoch * 1_000_000_000
+                + update.timeStamp.nanoseconds
+            )
+            if self._conn_status == ConnectionInfo.CONNECTED:
+                return None, None
+            elif self._conn_status == ConnectionInfo.NEVER_CONNECTED:
+                self._conn_status = ConnectionInfo.CONNECTED
+                return _serialise(self._source_name, self._conn_status, timestamp)
+        conn_state_map = {
+            Cancelled: ConnectionInfo.CANCELLED,
+            Disconnected: ConnectionInfo.DISCONNECTED,
+            RemoteError: ConnectionInfo.REMOTE_ERROR,
+            Finished: ConnectionInfo.FINISHED,
```

Comment:
I am thinking we should serialise all these as ConnectionInfo.DISCONNECTED.
But I am pretty sure we won't ever see anything other than disconnected, so it doesn't really matter how we treat them.

Reply:
I was trying to say it is not worth removing them because I don't think we will ever see them anyway.